### ISSUE TITLE
js form afterValidate executes only if form being submitted

### ISSUE THREAD

**Issue by yujin1st** [yiisoft/yii2#12080]
### What steps will reproduce the problem?

Setup before and afterValidate events for form, for example for blocking submit button:

``` javascript
  $('form').on('beforeValidate', function (e) {
    $(e.target).find(':submit').prop('disabled', true);
    console.log('beforeValidate');
  }).on('afterValidate', function (e) {
    $(e.target).find(':submit').prop('disabled', false);
    console.log('afterValidate');
  });
```
### What is the expected result?

Both event triggered
### What do you get instead?

Only beforeValidate event happens
### Additional info

In yii.activeForm.js afterValidate event described as it is triggered after validating the whole form.
But actually it happens only after validation triggered by form submission: see updateInputs method: it placed only if submitting is true
![2016-08-04 20 27 27](https://cloud.githubusercontent.com/assets/2317913/17400273/f580d464-5a81-11e6-89fb-d99af7242baf.png)

| Q | A |
| --- | --- |
| Yii version | 2.0.9 |
| PHP version | 5.6 |

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2/pull/12069 ?

**Comment by yujin1st** [comment]
Yes, sorry, didn't see that. 
As i understand, only way is to use previous version of framework, as new version will not be released soon?

**Comment by SilverFire** [comment]
Fixed by https://github.com/yiisoft/yii2/pull/12069

**Comment by czzplnm** [comment]
This is still the case.

**Comment by czzplnm** [comment]
Changing the updateInputs function of yii.activeForm.js to this works for me:

`        var errorAttributes = [];
        var $input = findInput($form, this);
        $.each(data.attributes, function () {
            if (!$input.is(":disabled") && !this.cancelled && updateInput($form, this, messages)) {
                errorAttributes.push(this);
            }
        });

        $form.trigger(events.afterValidate, [messages, errorAttributes]);
        updateSummary($form, messages);

        if (submitting) {`

**Comment by samdark** [comment]
@czzplnm how about making a pull request?

**Comment by czzplnm** [comment]
> @czzplnm how about making a pull request?

Done.

**Comment by mikehaertl** [comment]
This undid my fix for #17341 (validation broken in js strict mode) :cry: . PR to reapply the fix is here: #17398
